### DISCUSSION TITLE
How I can track % of cell is currently visible inside drawer controller

### DISCUSSION THREAD

**Discussion by SURYAKANTSHARMA** [scenee/FloatingPanel#654] [Q&A]
I can get % of cell visible in UIViewController like this 

```
extension UITableViewCell {
    func visiblePercentage(in superView: UIView) -> CGFloat {
        // Convert the cell's frame to the coordinate system of the table view
        let cellFrameInTable = self.convert(self.bounds, to: tableView)
        
        // Calculate the intersection between the cell's frame and the table view's bounds
        let visibleRect = cellFrameInTable.intersection(tableView.bounds)
        
        // If there's no intersection, the cell is not visible
        guard !visibleRect.isNull else { return 0 }
        
        // Calculate the percentage of the cell that is visible
        let visibleArea = visibleRect.width * visibleRect.height
        let totalArea = self.bounds.width * self.bounds.height
        
        return (visibleArea / totalArea) * 100
    }
}

```
It is working perfectly fine but while inside drawer vc I tried inside https://github.com/scenee/FloatingPanel/blob/master/Examples/Samples/Sources/ContentViewControllers/DebugTableViewController.swift is also not working I tried replacing
  let cellFrameInTable = self.convert(self.bounds, to: self.fpc!.surfaceView) and try to remove some offset but none of approach seems to working. 
  
  Can you guys please here ? 
  
  Let me any further question. 
  Attached sample working project without 
[DequeueOrNot-master 2.zip](https://github.com/user-attachments/files/18002506/DequeueOrNot-master.2.zip)

**Comment by SURYAKANTSHARMA** [comment]
Anyone looking into this ?

**Comment by SURYAKANTSHARMA** [comment]
@scenee Please help here

**Reply by scenee** [reply]
I looked over the code [DequeueOrNot-master 2.zip](https://github.com/user-attachments/files/18002506/DequeueOrNot-master.2.zip), but it doesn't use FloatingPanel 🤔 I suppose that whether `visiblePercentage(in:)` method works or not depends on how to layout a panel, which means what `FloatingPanelLayout` is used and which `FloatingPanelController.contentMode` is set.
